server/node/src/main/java/org/kaaproject/kaa/server/control/service/sdk/SchemaUtil.java

```diff
@@ -179,5 +182,20 @@ private static void parseChildSchemas(Schema parent, Map<String, Schema> namedSc
                 break;
         }
     }
+
+    public static Collection<JavaDynamicBean> compileAvroSchema(Schema avroSchema) {
+        try {
+            LOG.debug("Compiling {}", avroSchema);
+            Map<String, Schema> uniqueSchemas = SchemaUtil.getUniqueSchemasMap(Collections.singletonList(avroSchema));
+            List<JavaDynamicBean> javaSources = JavaSdkGenerator.generateSchemaSources(avroSchema, uniqueSchemas);
+            JavaDynamicCompiler compiler = new JavaDynamicCompiler();
+            compiler.init();
+            return compiler.compile(javaSources);
+        } catch (Exception cause) {
+            LOG.error(MessageFormat.format("Failed to compile {0}", avroSchema), cause);
+            String userMessage = "Failed to compile the schema. Please, ensure the schema body is correct";
```

Comment:
I think it is better to include cause. Smth like this:
String userMessage = "Failed to compile the schema: " + cause.getMessage();
In most cases, this will allow user to understand what is wrong.